src/fairchem/core/modules/normalizer.py

```diff
@@ -7,50 +7,190 @@
-        tensor: torch.Tensor | None = None,
-        mean=None,
-        std=None,
-        device=None,
-    ) -> None:
+        mean: float | torch.Tensor = 0.0,
+        std: float | torch.Tensor = 1.0,
+    ):
         """tensor is taken as a sample to calculate the mean and std"""
-        if tensor is None and mean is None:
-            return
+        super().__init__()
 
-        if device is None:
-            device = "cpu"
+        if isinstance(mean, float):
+            mean = torch.tensor(mean)
+        if isinstance(std, float):
+            std = torch.tensor(std)
 
-        self.mean: torch.Tensor
-        self.std: torch.Tensor
-        if tensor is not None:
-            self.mean = torch.mean(tensor, dim=0).to(device)
-            self.std = torch.std(tensor, dim=0).to(device)
-            return
-
-        if mean is not None and std is not None:
-            self.mean = torch.tensor(mean).to(device)
-            self.std = torch.tensor(std).to(device)
-
-    def to(self, device) -> None:
-        self.mean = self.mean.to(device)
-        self.std = self.std.to(device)
+        self.register_buffer(name="mean", tensor=mean)
+        self.register_buffer(name="std", tensor=std)
 
+    @torch.autocast(device_type="cuda", enabled=False)
     def norm(self, tensor: torch.Tensor) -> torch.Tensor:
         return (tensor - self.mean) / self.std
 
+    @torch.autocast(device_type="cuda", enabled=False)
     def denorm(self, normed_tensor: torch.Tensor) -> torch.Tensor:
         return normed_tensor * self.std + self.mean
 
-    def state_dict(self):
-        return {"mean": self.mean, "std": self.std}
+    def forward(self, normed_tensor: torch.Tensor) -> torch.Tensor:
+        return self.denorm(normed_tensor)
+
+    def load_state_dict(
+        self, state_dict: Mapping[str, Any], strict: bool = True, assign: bool = False
+    ):
+        # check if state dict is legacy state dicts
+        if isinstance(state_dict["mean"], float):
+            state_dict.update({k: torch.tensor(state_dict[k]) for k in ("mean", "std")})
+
+        super().load_state_dict(state_dict, strict=strict, assign=assign)
+
+
+def create_normalizer(
+    file: str | Path | None = None,
+    state_dict: dict | None = None,
+    tensor: torch.Tensor | None = None,
+    mean: float | torch.Tensor | None = None,
+    std: float | torch.Tensor | None = None,
+) -> Normalizer:
+    """Build a target data normalizers with optional atom ref
+
+    Args:
+        file (str or Path): path to pt or npz file.
+        state_dict (dict): a state dict for Normalizer module
+        tensor (Tensor): a tensor with target values used to compute mean and std
+        mean (float | Tensor): mean of target data
+        std (float | Tensor): std of target data
+        data_loader (DataLoader): a data loader used to estimate mean and std
+        num_batches (int): the number of batches used. If not given all batches are used.
+
+    Returns:
+        Normalizer
+    """
```

Comment:
Maybe this gets captured elsewhere - but slight concern that it's not transparent to the user if they define more than one of these in the config which actually gets used. Maybe hard error that only one needs to be defined? Open to alternatives.

Reply:
That's a good point. I'll address that here and in the trainer where this gets called.

Reply:
I log a warning in the create functions, but also raise an error in the base trainer if duplicates settings are given.